src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/Message.java

```diff
@@ -16,6 +19,20 @@
     private final HealthAuthorityNameAndAddress healthAuthorityNameAndAddress =
         HealthAuthorityNameAndAddress.fromString(extractSegment(HealthAuthorityNameAndAddress.KEY_QUALIFIER));
 
+    @Getter(lazy = true)
+    private final List<SG01> segmentGroup01s = SG01.createMultiple(getEdifactSegments().stream()
+        .dropWhile(segment -> !segment.startsWith(SG01.KEY))
+        .takeWhile(segment -> !segment.startsWith(SG02.KEY))
+        .takeWhile(segment -> !segment.startsWith(MessageTrailer.KEY))
+        .collect(toList()));
+
+    @Getter(lazy = true)
+    private final SG02 segmentGroup02 = new SG02(getEdifactSegments().stream()
+        .dropWhile(segment -> !segment.startsWith(SG02.KEY))
+        .skip(1)
+        .takeWhile(segment -> !segment.startsWith(MessageTrailer.KEY))
+        .collect(toList()));
```

Comment:
This approach is for when the section has a required single instance of a segment group as a direct child.

The `.dropWhile` condition describes how we know we're looking at the right segment group, and the `.takeWhile` condition(s) indicate signal(s) to stop. In this case, SG02 is not expecting to receive the `S02+02` segment so we skip it (but actually it wouldn't hurt anything if we did pass it in).